File: C++/145.binary-tree-postorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:

    // declaring a vector to store the answer
    vector<int> result;

    void traverse(TreeNode* root)
    {
        if (root == NULL) return;

        traverse(root->left);
        traverse(root->right);
        result.push_back(root->val);
    }


    vector<int> postorderTraversal(TreeNode* root) {
        traverse(root);
        return result;
    }
};

File: 145.binary-tree-postorder-traversal.py
# Definition for a binary tree node.
class TreeNode(object):
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution(object):
    def postorderTraversal(self, root):
        result = list()

        def traverse(root):
            if root is None: return

            traverse(root.left)
            traverse(root.right)
            result.append(root.val)

        traverse(root)
        return result